examples/src/main/scala/spinaldoc/examples/advanced/Slots.scala

```diff
@@ -0,0 +1,68 @@
+package spinaldoc.examples.advanced
+
+import spinal.core._
+import spinal.lib._
+import scala.language.postfixOps
+
+case class SlotsDemo(slotsCount : Int) extends Component {
+  //...
+  
+  
+  //Create the hardware for each slot
+  //Note each slot is an Area, not a Bundle
+  val slots = for(i <- 0 until slotsCount) yield new Area{
+    //Because the slot is an Area, we can define mix signal, registers, logic definitions
+    //Here are the registers for each slots
+    val valid = RegInit(False)
+    val address = Reg(UInt(8 bits))
+    val age = Reg(UInt(16 bits)) //Will count since how many cycles the slot is valid
+
+    //Here is some hardware behaviour for each slots
+    //Implement the age logic
+    when(valid){
+      age := age + 1
+    }
+
+    //removeIt will be used as a slot interface later on
+    val removeIt = False
+    when(removeIt){
+      valid := False
+    }
+  }
+
+  //Logic to allocate a new slot
+  val insert = new Area{
+    val cmd = Stream(UInt(8 bits)) //interface to issue requests
+    val free = slots.map(!_.valid)
+    val freeOh = OHMasking.first(free) //Get the first free slot (on hot mask)
+    cmd.ready := free.orR //Only allow cmd when there is a free slot
+    when(cmd.fire){
+      //slots.onMask(freeOh)(code) will execute the code for each slot where the corresponding freeOh bit is set
+      slots.onMask(freeOh){slot =>
```

Comment:
I think I'll add `onMask` to e.g. the tools, that warrants a separate explanation...

Reply:
Yes `onMask` and `onSel` deserve proper doc XD